### PR TITLE
review

### PR DESCRIPTION
**Pull request by EdwinKato**

### CODE REVIEW

File: space_allocator.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+dojo = Dojo()
+
+
+def docopt_cmd(func):
+    """
+    This decorator is used to simplify the try/except block and pass the result
+    of the docopt parsing to the called action.
+    """
+
+    def fn(self, arg):
```
Comment (faithngetich) [inline space_allocator.py:39]: It's best practice to clearly name your functions with descriptive names not initials

File: space_allocator.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    file = None
+
+    @docopt_cmd
+    def do_create_room(self, arg):
+        """Usage: create_room <room_type> <room_name>..."""
+
+        for room_name in arg['<room_name>']:
+            dojo.create_room(arg['<room_type>'], room_name)
+
+    def default(self, line):
```
Comment (faithngetich) [inline space_allocator.py:79]: If you decide to use docstring, use it consistently.

File: tests/test_dojo.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
+"""Unit tests for the application"""
+
```
Comment (faithngetich) [inline tests/test_dojo.py]: It's best practices to write your imports in alphabetical order.
e.g 
import colorful
import io
import unittest
from prettytable import PrettyTable
from src.dojo import Dojo

File: tests/test_dojo.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+import unittest
+import io
+import sys
+import colorful
+from prettytable import PrettyTable
+from src.dojo import Dojo
+
+
+class TestSpaceAllocator(unittest.TestCase):
```
Comment (faithngetich) [inline tests/test_dojo.py]: Just like in your previous files, your file name should give me an idea of what classes I should expect inside the file.
e.g In test_dojo.py , I expected to find a test_dojo.py class.

File: tests/test_dojo.py [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+        dojo.create_room("office", "orange")
+        dojo.reallocate_person(1, "orange")
+        target_room = dojo.find_room("orange")
+        person = dojo.find_person(1)
+        self.assertIn(person, target_room.residents)
+
+    def test_reallocate_if_person_had_no_living_space(self):
+        """Tests reallocate if person had no living space"""
+
+        dojo = Dojo()
```
Comment (faithngetich) [inline tests/test_dojo.py]: You probably need not to duplicate this instance and make use of what you already have in the setup.
Implement teardown . The setup and teardown methods are run before and after each test case.
Objects created during setup that are changed during one test are re-created for each test, so the tests don't effect each other.

File: tests/test_dojo.py [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+        """Tests that the application persists data"""
+
+        dojo1 = Dojo()
+        dojo1.create_room("office", "orange")
+        dojo1.add_person("John", "Ashaba", "Staff", "Y")
+        dojo1.save_state("mydb.db")
+        dojo2 = Dojo()
+        dojo2.load_state("mydb.db")
+        room = dojo2.find_room("orange")
+        self.assertIn(room, dojo2.all_rooms)
```
Comment (faithngetich) [inline tests/test_dojo.py]: Otherwise. Great Job 👍 !

Comment (EdwinKato) [reply tests/test_dojo.py]: Thank you..

**Pull request closed without merging**